packages/esm-patient-chart-app/src/visit-header/close-button.extension.tsx

```diff
@@ -0,0 +1,47 @@
+import { HeaderGlobalAction } from '@carbon/react';
+import { CloseFilledIcon, getHistory, goBackInHistory, navigate } from '@openmrs/esm-framework';
+import { usePatientChartStore } from '@openmrs/esm-patient-common-lib';
+import React, { useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import styles from './close-button.scss';
+
+function CloseButton() {
+  const { t } = useTranslation();
+  const { patientUuid } = usePatientChartStore();
+
+  const onClosePatientChart = useCallback(() => {
+    const history = getHistory();
+    // Get the last page the user was on before opening the patient chart by going backward
+    // through the history until a URL does not include patientUuid
+    let onCloseTarget = '';
+    for (let i = history.length - 1; i >= 0; i--) {
+      if (!history[i].includes(patientUuid)) {
+        onCloseTarget = history[i];
+        break;
+      }
+    }
+
+    if (onCloseTarget) {
+      goBackInHistory({ toUrl: onCloseTarget });
+    } else {
+      navigate({ to: '${openmrsSpaBase}/home' });
+    }
+  }, [patientUuid]);
+
+  const inPatientChart = window.location.pathname.includes(`${window.openmrsBase}/spa/patient/${patientUuid}`);
```

Comment:
The routing here is more complex because `${window.openmrsBase}/spa/patient/${patientUuid}/edit` is "in" the registration app.

Also, is preferable to use `window.getOpenmrsSpaBase()` as this ensures that the final `/` is always present. `window.openmrsBase` is a user supplied value that may or may not have a slash depedning on what someone entered. We also support the `spaBase` URL having no relation to `openmrsBase`.